packages/frontend/src/creation-form/index.tsx

```diff
@@ -184,32 +184,31 @@ export const Component = ({
             !isNaN(parseInt(minimumBond)) &&
             openingTimestamp.isAfter(dayjs())
         ) {
-            const formattedMinimumBond = ethers.utils.parseUnits(
-                minimumBond,
+            const formattedMinimumBond = parseUnits(
+                minimumBond as `${number}`,
                 chain.nativeCurrency.decimals
             );
             initializationDataGetter = async () => {
                 const questionCid = await uploadToIpfs(question);
-                const initializationData = ethers.utils.defaultAbiCoder.encode(
+                const initializationData = encodeAbiParameters(
                     [
-                        "address",
-                        "uint256",
-                        "string",
-                        "uint32",
-                        "uint32",
-                        "uint256",
+                        { type: "address", name: "arbitratorAddress" },
+                        { type: "uint256", name: "realityTemplateId" },
+                        { type: "string", name: "questionIdentifier" },
+                        { type: "uint32", name: "questionTimeout" },
+                        { type: "uint32", name: "openingTimestamp" },
+                        { type: "uint256", name: "minimumBond" },
                     ],
                     [
-                        arbitrator.value,
-                        realityTemplateId.value,
+                        arbitrator.value as `0x${string}`,
+                        BigInt(realityTemplateId.value),
```

Comment:
I think with some of these it probably makes sense to convert the source type to bigint instead of casting.

Reply:
I'll try to do this when possible, in this case, the `.value` type comes from the `ValueType` inside `SelectOption` (imported from the `UI` package), so it needs to be casted.

Reply:
Oh, I understand. Isn't the `Select` in the UI lib typed in such a way to respect the type of the options given to it?

Reply:
Absolutely, we should.